service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementMapper.java

```diff
@@ -0,0 +1,223 @@
+    private static final Mustache MEDICATION_STATEMENT_PRESCRIBE_TEMPLATE =
+        TemplateUtils.loadTemplate("ehr_medication_statement_prescribe_template.mustache");
+
+    private static final String ACTIVE_STATUS_CODE = "ACTIVE";
+    private static final String COMPLETE_STATUS_CODE = "COMPLETE";
+    private static final String NOTES = "Notes: %s";
+    private static final String EXPECTED_SUPPLY_DURATION = "Expected Supply Duration: %s %s";
+    private static final String PATIENT_INSTRUCTION = "Patient Instruction: %s";
+    private static final String DEFAULT_QUANTITY_VALUE = "1";
+    private static final String DEFAULT_QUANTITY_TEXT = "Unk UoM";
+    private static final List<String> ACUTE_PRESCRIPTION_TYPE_CODES = Arrays.asList("acute", "acute-handwritten");
+    private static final List<String> REPEAT_PRESCRIPTION_TYPE_CODES =
+        Arrays.asList("delayed-prescribing", "repeat", "repeat-dispensing");
+    private static final String ACUTE_REPEAT_VALUE = "0";
+
+    private final MessageContext messageContext;
+    private final CodeableConceptCdMapper codeableConceptCdMapper;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+
+    public String mapMedicationRequestToMedicationStatement(MedicationRequest medicationRequest) {
+        var medicationStatementTemplateParameters = MedicationStatementTemplateParameters.builder()
+            .medicationStatementId(messageContext.getIdMapper().getOrNew(ResourceType.MedicationRequest, medicationRequest.getId()))
+            .statusCode(buildStatusCode(medicationRequest))
+            .effectiveTime(StatementTimeMappingUtils.prepareEffectiveTimeForMedicationRequest(medicationRequest))
+            .availabilityTime(StatementTimeMappingUtils.prepareAvailabilityTimeForMedicationRequest(medicationRequest))
+            .medicationReferenceCode(buildMedicationReferenceCode(medicationRequest))
+            .ehrSupplyId(messageContext.getMedicationRequestIdMapper().getOrNew(medicationRequest.getId()))
+            .medicationStatementPertinentInformation(buildDosageInstructionPertinentInformation(medicationRequest))
+            .ehrSupplyPertinentInformation(buildPertinentInformation(medicationRequest))
+            .repeatNumber(buildRepeatValue(medicationRequest))
+            .quantityValue(buildQuantityValue(medicationRequest))
+            .quantityText(buildQuantityText(medicationRequest))
+            .hasEhrSupplyDiscontinue(hasStatusReasonStopped(medicationRequest))
+            .ehrSupplyDiscontinueCode(buildStatusReasonCode(medicationRequest))
+            .ehrSupplyDiscontinueId(randomIdGeneratorService.createNewId())
+            .ehrSupplyDiscontinueAvailabilityTime(buildStatusReasonAvailabilityTime(medicationRequest))
+            .priorPrescriptionId(buildPriorPrescription(medicationRequest))
+            .basedOn(buildBasedOn(medicationRequest))
+            .build();
```

Comment:
This might be easier to read if we separate the derivation of values from building the object. There'll be more lines, but we're not playing code golf and the reader will quickly see what's going on and skim over the object building. What do you think?